providers/common/exchange/dh_exch.c

```diff
@@ -42,22 +42,28 @@ static int dh_init(void *vpdhctx, void *vdh)
 {
     PROV_DH_CTX *pdhctx = (PROV_DH_CTX *)vpdhctx;
 
+    if (pdhctx == NULL)
+        return 0;
     DH_free(pdhctx->dh);
     pdhctx->dh = vdh;
+    if (vdh == NULL)
+        return 0;
     DH_up_ref(pdhctx->dh);
```

Comment:
Please check the return code of DH_up_ref

Reply:
Done.